src/Components/PollItems.jsx

```diff
@@ -0,0 +1,60 @@
+
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { submitVote } from '../slices/pollsSlice';
+// Assuming these constants are defined somewhere in your project
+import { ADMIN_ID, HR_ID } from '../utils/constantData';
+
+const PollItem = ({ poll }) => {
+  const dispatch = useDispatch();
+  const userVotes = useSelector((state) => state.polls.userVotes);
+  const roleId = useSelector((state) => state.auth.roleId); 
+  const hasVoted = userVotes[poll.id] !== undefined;
+
+  const handleVote = (voteOption) => {
+    if (!hasVoted) {
+      dispatch(submitVote({ pollId: poll.id, voteOption }));
+    }
+  };
+
+  // Check if the user is admin or HR
+  const isAdmin = roleId === ADMIN_ID || roleId === HR_ID;
+
+  return (
+    <div className="poll-item p-4 border rounded-lg shadow-lg">
+      <h3 className="font-bold text-lg">{poll.title}</h3>
+      <div className="options mt-2 space-x-2">
+        {poll.options.map((option, index) => (
+          <button
+            key={index}
+            onClick={() => handleVote(option)}
+            disabled={hasVoted}
+            className={`${
+              hasVoted ? 'bg-gray-400' : 'bg-blue-500 hover:bg-blue-600'
+            } text-white py-1 px-3 rounded`}
+          >
+            {option} {hasVoted && option === userVotes[poll.id] ? '(Voted)' : ''}
+          </button>
+        ))}
+      </div>
+      {isAdmin && (
```

Comment:
try to create multiple functional components, which should return a view based on some condition. It will clear the code and make it more readable.